### ISSUE TITLE
[crash] Errors when opening dap-ui with treesitter and then crash on close

### ISSUE THREAD

**Issue by Smjert** [neovim/neovim#19989]
### Neovim version (nvim -v)

v0.7.2, v0.8.0-dev-899-gff5cfcdea

### Vim (not Nvim) behaves the same?

Setup is nvim specific

### Operating system/version

ArchLinux

### Terminal name/version

Tilix 1.9.5

### $TERM environment variable

xterm-256-color

### Installation

built from repo

### How to reproduce the issue

I tried to reduce the steps and parts involved, but my knowledge of neovim internals is basically nonexistent and I could only reproduce it with this weird combination:

1. Configure neovim with the following setting
```
vim.o.cmdheight=2
```
2. Install `packer` at commit `90b323bccc04ad9b23c971a85813a1405c7725a8`
3. Install `nvim-treesitter` using packer:
```
local use = require('packer').use
[...]
use { 'nvim-treesitter/nvim-treesitter',
    run = function() require('nvim-treesitter.install').update({ with_sync = true }) end,
    commit = '9560e3ce7d9f01b21f7f14c0b2d93f718e1ccf36'
}
```
and use the following configuration:
```
require'nvim-treesitter.configs'.setup {
  ensure_installed = {'cpp'},
}

vim.api.nvim_create_autocmd({'BufEnter','BufAdd','BufNew','BufNewFile','BufWinEnter'}, {
  group = vim.api.nvim_create_augroup('TS_FOLD_WORKAROUND', {}),
  callback = function()
    vim.opt.foldmethod     = 'expr'
    vim.opt.foldexpr       = 'nvim_treesitter#foldexpr()'
  end
})
```
3. Ensure that `nvim-treesitter` has compiled and installed the cpp parser with `:TSUpdate` and then `:TSInstallInfo`
4. Install `nvim-dap` and `nvim-dap-ui`:
```
use { 'mfussenegger/nvim-dap',
    tag = '0.2.0'
}
use { 'rcarriga/nvim-dap-ui',
    commit = '225115ae986b39fdaffaf715e571dd43b3ac9670'
}
```
configure `nvim-dap-ui` with:
```
require('dapui').setup()
```
5. Install `nvim-cmp` and `cmp-cmdline` (or `cmp-path`)
```
use { 'hrsh7th/cmp-cmdline',
    commit = '9c0e331fe78cab7ede1c051c065ee2fc3cf9432e'
}
use { 'hrsh7th/nvim-cmp',
    commit = '706371f1300e7c0acb98b346f80dad2dd9b5f679'
}
```
6. Open the following C file:
```
#include <stdio.h>

struct Test {
    int a;
}

int main() {

    char c[10];

    c[1] = 'a';
    c[1] = 'a';
    c[1] = 'a';
    c[1] = 'a';
    c[1] = 'a';
    c[1] = 'a';
    c[1] = 'a';
    c[1] = 'a';
    c[1] = 'a';
    c[1] = 'a';
    c[1] = 'a';
    c[1] = 'a';
    c[1] = 'a';
    c[1] = 'a';

    return 1;
}
```
7. Move the cursor to the fourth line, where `int a;` is
8. Open the dap-ui giving the command `:lua require('dapui').open()`
9. Try to move the cursor
10. Close with `:qa`

### Expected behavior

I would not expect to have errors at step 9, and at step109 I would not expect neovim to crash.

### Actual behavior

At step 9 I get a couple of repeated errors, while moving the cursor up:
```
E315: ml_get: invalid lnum:3
```
And then at step 10 I get a crash with `free(): invalid pointer`

**Comment by Smjert** [comment]
A couple of additional notes: the culprit seems to be cause by that auto command for the folds.

Below a stacktrace taken from a Debug build I did of v0.8.0-dev-899-gff5cfcdea:
```
Thread 2 (Thread 0x7ffff764c6c0 (LWP 83236) "nvim"):
#0  0x00007ffff7c5fd06 in epoll_wait (epfd=15, events=0x7ffff76485f0, maxevents=1024, timeout=-1) at ../sysdeps/unix/sysv/linux/epoll_wait.c:30
#1  0x00007ffff7f4fe1f in ?? () from /usr/lib/libuv.so.1
#2  0x00007ffff7f3954a in uv_run () from /usr/lib/libuv.so.1
#3  0x00005555556ee1eb in loop_poll_events (loop=0x7ffff764b7e8, ms=-1) at /home/smjert/Development/neovim/src/nvim/event/loop.c:61
#4  0x0000555555930a59 in tui_main (bridge=0x555555ada5d0, ui=0x555555ada490) at /home/smjert/Development/neovim/src/nvim/tui/tui.c:505
#5  0x0000555555943810 in ui_thread_run (data=0x555555ada5d0) at /home/smjert/Development/neovim/src/nvim/ui_bridge.c:106
#6  0x00007ffff7bde78d in start_thread (arg=<optimized out>) at pthread_create.c:442
#7  0x00007ffff7c5f8e4 in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:100

Thread 1 (Thread 0x7ffff7976780 (LWP 83220) "nvim"):
#0  __pthread_kill_implementation (threadid=<optimized out>, signo=signo@entry=6, no_tid=no_tid@entry=0) at pthread_kill.c:44
#1  0x00007ffff7be0543 in __pthread_kill_internal (signo=6, threadid=<optimized out>) at pthread_kill.c:78
#2  0x00007ffff7b90998 in __GI_raise (sig=sig@entry=6) at ../sysdeps/posix/raise.c:26
#3  0x00007ffff7b7a53d in __GI_abort () at abort.c:79
#4  0x00007ffff7bd467e in __libc_message (action=action@entry=do_abort, fmt=fmt@entry=0x7ffff7cf3433 "%s\n") at ../sysdeps/posix/libc_fatal.c:155
#5  0x00007ffff7bea26c in malloc_printerr (str=str@entry=0x7ffff7cf1098 "free(): invalid pointer") at malloc.c:5660
#6  0x00007ffff7bec0bc in _int_free (av=<optimized out>, p=<optimized out>, have_lock=have_lock@entry=0) at malloc.c:4435
#7  0x00007ffff7beea33 in __GI___libc_free (mem=<optimized out>) at malloc.c:3385
#8  0x00005555557d9d85 in xfree (ptr=0x555555ab7090) at /home/smjert/Development/neovim/src/nvim/memory.c:125
#9  0x0000555555911006 in reset_synblock (wp=0x555555ab1b30) at /home/smjert/Development/neovim/src/nvim/syntax.c:3246
#10 0x0000555555988611 in win_close_buffer (win=0x555555ab1b30, free_buf=false, abort_if_last=true) at /home/smjert/Development/neovim/src/nvim/window.c:2639
#11 0x000055555597e9bb in win_close (win=0x555555ab1b30, free_buf=false, force=false) at /home/smjert/Development/neovim/src/nvim/window.c:2807
#12 0x0000555555641e1e in do_buffer (action=4, start=1, dir=1, count=3, forceit=1) at /home/smjert/Development/neovim/src/nvim/buffer.c:1268
#13 0x00005555555ed59e in nvim_buf_delete (buffer=3, opts=..., err=0x7fffffffc518) at /home/smjert/Development/neovim/src/nvim/api/buffer.c:1117
#14 0x00005555555d7795 in nlua_api_nvim_buf_delete (lstate=0x7ffff7fa2380) at src/nvim/auto/lua_api_c_bindings.generated.c:1147
#15 0x00007ffff7d48ef6 in lj_BC_FUNCC () at buildvm_x86.dasc:857
#16 0x00007ffff7d5b873 in lua_pcall (L=0x7ffff7fa2380, nargs=<optimized out>, nresults=0, errfunc=<optimized out>) at /build/luajit/src/luajit-2.0-633f265/src/lj_api.c:1116
#17 0x0000555555791bd2 in nlua_pcall (lstate=0x7ffff7fa2380, nargs=0, nresults=0) at /home/smjert/Development/neovim/src/nvim/lua/executor.c:135
#18 0x0000555555791948 in nlua_typval_exec (lcmd=0x555555ca07e0 "require('dap')._vim_exit_handler()", lcmd_len=34, name=0x5555559c16f5 ":lua", args=0x0, argcount=0, special=false, ret_tv=0x0) at /home/smjert/Development/neovim/src/nvim/lua/executor.c:1305
#19 0x00005555557928c8 in ex_lua (eap=0x7fffffffc7e0) at /home/smjert/Development/neovim/src/nvim/lua/executor.c:1497
#20 0x000055555570a9a2 in execute_cmd0 (retv=0x7fffffffc7dc, eap=0x7fffffffc7e0, errormsg=0x7fffffffc898, preview=false) at /home/smjert/Development/neovim/src/nvim/ex_docmd.c:1570
#21 0x0000555555706b3d in do_one_cmd (cmdlinep=0x7fffffffcb78, flags=7, cstack=0x7fffffffcb80, fgetline=0x55555563d010 <getnextac>, cookie=0x7fffffffd148) at /home/smjert/Development/neovim/src/nvim/ex_docmd.c:2219
#22 0x0000555555704477 in do_cmdline (cmdline=0x0, fgetline=0x55555563d010 <getnextac>, cookie=0x7fffffffd148, flags=7) at /home/smjert/Development/neovim/src/nvim/ex_docmd.c:572
#23 0x000055555563bb42 in apply_autocmds_group (event=EVENT_EXITPRE, fname=0x555555b611a0 "/home/smjert/Development/test-cmake/[dap-repl]", fname_io=0x0, force=false, group=-3, buf=0x555555c47980, eap=0x0, data=0x0) at /home/smjert/Development/neovim/src/nvim/autocmd.c:1861
#24 0x000055555563ca53 in apply_autocmds (event=EVENT_EXITPRE, fname=0x0, fname_io=0x0, force=false, buf=0x555555c47980) at /home/smjert/Development/neovim/src/nvim/autocmd.c:1527
#25 0x000055555571008e in before_quit_autocmds (wp=0x555555ab1b30, quit_all=true, forceit=false) at /home/smjert/Development/neovim/src/nvim/ex_docmd.c:5196
#26 0x00005555557154df in ex_quit_all (eap=0x7fffffffd400) at /home/smjert/Development/neovim/src/nvim/ex_docmd.c:5301
#27 0x000055555570a9a2 in execute_cmd0 (retv=0x7fffffffd3fc, eap=0x7fffffffd400, errormsg=0x7fffffffd4b8, preview=false) at /home/smjert/Development/neovim/src/nvim/ex_docmd.c:1570
#28 0x0000555555706b3d in do_one_cmd (cmdlinep=0x7fffffffd798, flags=0, cstack=0x7fffffffd7a0, fgetline=0x55555571e000 <getexline>, cookie=0x0) at /home/smjert/Development/neovim/src/nvim/ex_docmd.c:2219
#29 0x0000555555704477 in do_cmdline (cmdline=0x0, fgetline=0x55555571e000 <getexline>, cookie=0x0, flags=0) at /home/smjert/Development/neovim/src/nvim/ex_docmd.c:572
#30 0x0000555555807edd in nv_colon (cap=0x7fffffffdde0) at /home/smjert/Development/neovim/src/nvim/normal.c:4050
#31 0x0000555555805a14 in normal_execute (state=0x7fffffffdd60, key=58) at /home/smjert/Development/neovim/src/nvim/normal.c:1170
#32 0x0000555555902678 in state_enter (s=0x7fffffffdd60) at /home/smjert/Development/neovim/src/nvim/state.c:88
#33 0x00005555557feb44 in normal_enter (cmdwin=false, noexmode=false) at /home/smjert/Development/neovim/src/nvim/normal.c:469
#34 0x000055555579e213 in main (argc=2, argv=0x7fffffffe1c8) at /home/smjert/Development/neovim/src/nvim/main.c:585
```

And ASAN in RelWithDebInfo (NOTE the stack trace of the allocation wasn't always the same):
```
==76691==ERROR: AddressSanitizer: attempting free on address which was not malloc()-ed: 0x627000002b80 in thread T0
    #0 0x56496ae56702 in __interceptor_free.part.0 asan_malloc_linux.cpp.o
    #1 0x56496b653786 in xfree /home/smjert/Development/neovim/src/nvim/memory.c:125:3
    #2 0x56496b653786 in reset_synblock /home/smjert/Development/neovim/src/nvim/syntax.c:3246:5
    #3 0x56496b75999f in win_close_buffer /home/smjert/Development/neovim/src/nvim/window.c:2639:5
    #4 0x56496b73dd7d in win_close /home/smjert/Development/neovim/src/nvim/window.c:2807:3
    #5 0x56496af942eb in do_buffer /home/smjert/Development/neovim/src/nvim/buffer.c:1268:11
    #6 0x56496aed108e in nvim_buf_delete /home/smjert/Development/neovim/src/nvim/api/buffer.c:1117:16
    #7 0x56496aea30fd in nlua_api_nvim_buf_delete /home/smjert/Development/neovim/build/src/nvim/auto/lua_api_c_bindings.generated.c:1147:3
    #8 0x7f74150daef5  (/usr/lib/libluajit-5.1.so.2+0x9ef5) (BuildId: 2c55b40fcc447be9d6e78693b381f17f4ca69316)
    #9 0x7f74150ed872 in lua_pcall (/usr/lib/libluajit-5.1.so.2+0x1c872) (BuildId: 2c55b40fcc447be9d6e78693b381f17f4ca69316)
    #10 0x56496b2c85c2 in nlua_pcall /home/smjert/Development/neovim/src/nvim/lua/executor.c:135:16
    #11 0x56496b2c814d in nlua_typval_exec /home/smjert/Development/neovim/src/nvim/lua/executor.c:1305:7
    #12 0x56496b2c9fdd in ex_lua /home/smjert/Development/neovim/src/nvim/lua/executor.c:1497:3
    #13 0x56496b188249 in execute_cmd0 /home/smjert/Development/neovim/src/nvim/ex_docmd.c:1570:7
    #14 0x56496b1799db in do_one_cmd /home/smjert/Development/neovim/src/nvim/ex_docmd.c:2219:7
    #15 0x56496b1799db in do_cmdline /home/smjert/Development/neovim/src/nvim/ex_docmd.c:572:20
    #16 0x56496af838d5 in apply_autocmds_group /home/smjert/Development/neovim/src/nvim/autocmd.c:1861:5
    #17 0x56496b193d20 in apply_autocmds /home/smjert/Development/neovim/src/nvim/autocmd.c:1527:10
    #18 0x56496b193d20 in before_quit_autocmds /home/smjert/Development/neovim/src/nvim/ex_docmd.c:5196:5
    #19 0x56496b1a0c38 in ex_quit_all /home/smjert/Development/neovim/src/nvim/ex_docmd.c:5301:7
    #20 0x56496b188249 in execute_cmd0 /home/smjert/Development/neovim/src/nvim/ex_docmd.c:1570:7
    #21 0x56496b1799db in do_one_cmd /home/smjert/Development/neovim/src/nvim/ex_docmd.c:2219:7
    #22 0x56496b1799db in do_cmdline /home/smjert/Development/neovim/src/nvim/ex_docmd.c:572:20
    #23 0x56496b3f3253 in nv_colon /home/smjert/Development/neovim/src/nvim/normal.c:4050:20
    #24 0x56496b3e7802 in normal_execute /home/smjert/Development/neovim/src/nvim/normal.c:1170:3
    #25 0x56496b632310 in state_enter /home/smjert/Development/neovim/src/nvim/state.c:88:26
    #26 0x56496b3d6987 in normal_enter /home/smjert/Development/neovim/src/nvim/normal.c:469:3
    #27 0x56496b2e363b in main /home/smjert/Development/neovim/src/nvim/main.c:585:3
    #28 0x7f7414eed2cf  (/usr/lib/libc.so.6+0x232cf) (BuildId: 9c28cfc869012ebbd43cdb0f1eebcd14e1b8bdd8)
    #29 0x7f7414eed389 in __libc_start_main (/usr/lib/libc.so.6+0x23389) (BuildId: 9c28cfc869012ebbd43cdb0f1eebcd14e1b8bdd8)
    #30 0x56496ada2844 in _start /build/glibc/src/glibc/csu/../sysdeps/x86_64/start.S:115

0x627000002b80 is located 10880 bytes inside of 12320-byte region [0x627000000100,0x627000003120)
allocated by thread T0 here:
    #0 0x56496ae57a71 in __interceptor_calloc (/usr/local/bin/nvim+0x650a71)
    #1 0x56496b376de2 in xcalloc /home/smjert/Development/neovim/src/nvim/memory.c:139:15
    #2 0x56496af95f0b in buflist_new /home/smjert/Development/neovim/src/nvim/buffer.c:1766:11
    #3 0x56496b75d212 in win_alloc_firstwin /home/smjert/Development/neovim/src/nvim/window.c:3962:14
    #4 0x56496b75cd7e in win_alloc_first /home/smjert/Development/neovim/src/nvim/window.c:3922:7
    #5 0x56496b2df749 in early_init /home/smjert/Development/neovim/src/nvim/main.c:192:8
    #6 0x56496b2dfea0 in main /home/smjert/Development/neovim/src/nvim/main.c:259:3
    #7 0x7f7414eed2cf  (/usr/lib/libc.so.6+0x232cf) (BuildId: 9c28cfc869012ebbd43cdb0f1eebcd14e1b8bdd8)
 ```

**Comment by Smjert** [comment]
I had to edit the report because while looking at this again today I realized there were more things at play, because I didn't mention packer (if only in the crash/asan comment) and I forgot I was still loading other plugins because I did not remove their `use` block.

So there are 2 new plugins to install, but not require/configure.
The crash seems to be introduced by `nvim-cmp` and one of it's additional plugins.

Basically when I only install `nvim-cmp` or I don't install it, I still get the ml_get errors when moving the cursor, but there's no crash on `:qa`; instead this only closes the current buffer where the `main.c` file is, and the other buffers from `dap-ui` stays open, and I have to give a second `:qa` to close.

It's only when I install `nvim-cmp` + `cmp-cmdline` or `cmp-path` that the crash happens.

**Comment by Smjert** [comment]
Debugging this a little seems that:
```c
/*
 * Get rid of ownsyntax for window "wp".
 */
void reset_synblock(win_T *wp)
{
  if (wp->w_s != &wp->w_buffer->b_s) { <----
    syntax_clear(wp->w_s);
    xfree(wp->w_s);
    wp->w_s = &wp->w_buffer->b_s;
  }
}
```
that check should in theory prevent incorrectly freeing the syntax block `file_buffer` struct member, which is not allocate on its own, but the problem is that the syntax block the window `w_s` member doesn't point to the current buffer in the window, so that checks passes, although `w_s` doesn't point to something allocated.

So it seems `w_s` points to an old buffer syntax block and got desynced from the current buffer?

**Comment by Smjert** [comment]
@zeertzjq thanks for the heads up!

I still get the crash on NVIM v0.9.0-dev-488+g8b84a10db
```
Thread 2 (Thread 0x7ffff764f6c0 (LWP 59947) "nvim"):
#0  0x00007ffff7c5e096 in epoll_wait (epfd=15, events=0x7ffff764b5b0, maxevents=1024, timeout=-1) at ../sysdeps/unix/sysv/linux/epoll_wait.c:30
#1  0x00007ffff7f4fe1f in ?? () from /usr/lib/libuv.so.1
#2  0x00007ffff7f3954a in uv_run () from /usr/lib/libuv.so.1
#3  0x00005555556db168 in loop_uv_run (loop=0x7ffff764e7e0, once=true, ms=<optimized out>) at /home/smjert/Development/neovim/src/nvim/event/loop.c:65
#4  loop_poll_events (loop=0x7ffff764e7e0, ms=<optimized out>) at /home/smjert/Development/neovim/src/nvim/event/loop.c:87
#5  0x00005555558e2cdd in tui_main (bridge=0x555555a76650, ui=0x555555a79020) at /home/smjert/Development/neovim/src/nvim/tui/tui.c:509
#6  0x00007ffff7bdc8fd in start_thread (arg=<optimized out>) at pthread_create.c:442
#7  0x00007ffff7c5ea60 in clone3 () at ../sysdeps/unix/sysv/linux/x86_64/clone3.S:81

Thread 1 (Thread 0x7ffff7979780 (LWP 59941) "nvim"):
#0  __pthread_kill_implementation (threadid=<optimized out>, signo=signo@entry=6, no_tid=no_tid@entry=0) at pthread_kill.c:44
#1  0x00007ffff7bde6b3 in __pthread_kill_internal (signo=6, threadid=<optimized out>) at pthread_kill.c:78
#2  0x00007ffff7b8e958 in __GI_raise (sig=sig@entry=6) at ../sysdeps/posix/raise.c:26
#3  0x00007ffff7b7853d in __GI_abort () at abort.c:79
#4  0x00007ffff7bd27ee in __libc_message (action=action@entry=do_abort, fmt=fmt@entry=0x7ffff7cf144b "%s\n") at ../sysdeps/posix/libc_fatal.c:155
#5  0x00007ffff7be83dc in malloc_printerr (str=str@entry=0x7ffff7cef0b1 "free(): invalid pointer") at malloc.c:5660
#6  0x00007ffff7bea22c in _int_free (av=<optimized out>, p=<optimized out>, have_lock=have_lock@entry=0) at malloc.c:4435
#7  0x00007ffff7becba3 in __GI___libc_free (mem=<optimized out>) at malloc.c:3385
#8  0x000055555591a849 in xfree (ptr=0xea25) at /home/smjert/Development/neovim/src/nvim/memory.c:136
#9  reset_synblock (wp=0x555555a4fb40) at /home/smjert/Development/neovim/src/nvim/syntax.c:3016
#10 win_close_buffer (win=0x555555a4fb40, free_buf=false, abort_if_last=true) at /home/smjert/Development/neovim/src/nvim/window.c:2627
#11 0x0000555555912846 in win_close (win=0x555555a4fb40, free_buf=188, force=<optimized out>) at /home/smjert/Development/neovim/src/nvim/window.c:2794
#12 0x0000555555630632 in do_buffer (action=4, start=<optimized out>, dir=<optimized out>, count=<optimized out>, forceit=1) at /home/smjert/Development/neovim/src/nvim/buffer.c:1334
#13 0x00005555555f0a58 in nvim_buf_delete (buffer=<optimized out>, opts=..., err=0x7fffffffc730) at /home/smjert/Development/neovim/src/nvim/api/buffer.c:1130
#14 0x00005555555d900d in nlua_api_nvim_buf_delete (lstate=0x7ffff7fa4380) at src/nvim/auto/lua_api_c_bindings.generated.c:1154
#15 0x00007ffff7d46ef6 in lj_BC_FUNCC () at buildvm_x86.dasc:857
#16 0x00007ffff7d598b3 in lua_pcall (L=0x7ffff7fa4380, nargs=<optimized out>, nresults=0, errfunc=<optimized out>) at /usr/src/debug/luajit/luajit-2.0-637721d/src/lj_api.c:1116
#17 0x000055555576acad in nlua_pcall (lstate=0x7ffff7fa4380, nargs=0, nresults=<optimized out>) at /home/smjert/Development/neovim/src/nvim/lua/executor.c:153
#18 nlua_typval_exec (lcmd=<optimized out>, lcmd_len=<optimized out>, name=<optimized out>, args=<optimized out>, argcount=0, special=<optimized out>, ret_tv=0x0) at /home/smjert/Development/neovim/src/nvim/lua/executor.c:1396
#19 0x000055555576bc78 in ex_lua (eap=<optimized out>) at /home/smjert/Development/neovim/src/nvim/lua/executor.c:1591
#20 0x00005555556f542a in execute_cmd0 (retv=0x7fffffffcabc, eap=0x7fffffffc9f0, errormsg=0x7fffffffcab0, preview=false) at /home/smjert/Development/neovim/src/nvim/ex_docmd.c:1621
#21 0x00005555556f0903 in do_one_cmd (cmdlinep=0x7fffffffc9e8, flags=7, cstack=0x7fffffffcac0, fgetline=0x55555562a4b0 <getnextac>, cookie=0x7fffffffd160) at /home/smjert/Development/neovim/src/nvim/ex_docmd.c:2278
#22 do_cmdline (cmdline=<optimized out>, fgetline=0x55555562a4b0 <getnextac>, cookie=0x7fffffffd160, flags=7) at /home/smjert/Development/neovim/src/nvim/ex_docmd.c:579
#23 0x0000555555628d7b in apply_autocmds_group (event=EVENT_EXITPRE, fname=<optimized out>, fname_io=<optimized out>, force=<optimized out>, group=<optimized out>, buf=0x555555b97b30, eap=0x0, data=0x0) at /home/smjert/Development/neovim/src/nvim/autocmd.c:1924
#24 0x00005555556f7e3b in apply_autocmds (event=EVENT_EXITPRE, fname=0x0, fname_io=0x0, force=false, buf=0x555555b97b30) at /home/smjert/Development/neovim/src/nvim/autocmd.c:1590
#25 before_quit_autocmds (wp=0x555555a4fb40, quit_all=<optimized out>, forceit=<optimized out>) at /home/smjert/Development/neovim/src/nvim/ex_docmd.c:4469
#26 0x00005555556fd66c in ex_quit_all (eap=0x7fffffffd430) at /home/smjert/Development/neovim/src/nvim/ex_docmd.c:4574
#27 0x00005555556f542a in execute_cmd0 (retv=0x7fffffffd4fc, eap=0x7fffffffd430, errormsg=0x7fffffffd4f0, preview=false) at /home/smjert/Development/neovim/src/nvim/ex_docmd.c:1621
#28 0x00005555556f0903 in do_one_cmd (cmdlinep=0x7fffffffd428, flags=0, cstack=0x7fffffffd500, fgetline=0x555555705d40 <getexline>, cookie=0x0) at /home/smjert/Development/neovim/src/nvim/ex_docmd.c:2278
#29 do_cmdline (cmdline=<optimized out>, fgetline=0x555555705d40 <getexline>, cookie=0x0, flags=0) at /home/smjert/Development/neovim/src/nvim/ex_docmd.c:579
#30 0x00005555557d5fe7 in nv_colon (cap=0x7fffffffdf40) at /home/smjert/Development/neovim/src/nvim/normal.c:3213
#31 0x00005555557d280f in normal_execute (state=0x7fffffffdec0, key=<optimized out>) at /home/smjert/Development/neovim/src/nvim/normal.c:1180
#32 0x00005555558ad64a in state_enter (s=0x7fffffffdec0) at /home/smjert/Development/neovim/src/nvim/state.c:99
#33 0x000055555577ad4c in normal_enter (cmdwin=false, noexmode=false) at /home/smjert/Development/neovim/src/nvim/normal.c:477
#34 main (argc=2, argv=<optimized out>) at /home/smjert/Development/neovim/src/nvim/main.c:595
```

**Comment by zeertzjq** [comment]
https://github.com/vim/vim/issues/11687
